src/compat.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of the Black Magic Debug project.
+ *
+ * Copyright (C) 2019-2020 Uwe Bonnes <bon@elektron.ikp.physik.tu-darmstadt.de>
```

Comment:
Please use the copyright notice from sfdp.c here, with the year updated to 2023 - the ulog2 function is code we contributed in #1464 (this should also serve as a reminder to ourself to fix the stlinkv2.c header).

We'd ask you to update the license terms to the same as sfdp.c but that's entirely optional.

Reply:
Sorry, somehow messed up using `git blame` and traced it back to when `stlink.c` was first introduced. Couldn't find it in the other referenced projects so I incorrectly attributed it to Uwe Bonnes. Fixed now.

Reply:
No worries :) it happens.